config/environments/development.rb

```diff
@@ -67,4 +67,8 @@
 
   # Uncomment if you wish to allow Action Cable access from any origin.
   # config.action_cable.disable_request_forgery_protection = true
+
+  # Use default Rails logging
+  config.log_formatter = ::Logger::Formatter.new
+  config.lograge.enabled = false
```

Comment:
I agree the change is needed, I'm just thinking that the default should always be Rails Logger rather than the exception. So instead of declaring the rails logger here, we should only declare and set Lograge in production as that is the exception

Reply:
But instead of all the config in application.rb could we not just move it to production.rb and then if you want to test it you can run rails in production mode to see what the logs will output?

Reply:
I meant test in specs, if we have the logging configuration in `application.rb` then we don't need to duplicate it between `environments/production.rb` and `environments/test.rb`.

Reply:
Overriding the log formatter in the dev config was how I'd anticipated it being managed if the json output wasn't clear enough for local dev where the logs are viewed without json parsing power. I don't think having to run the apps in Production mode is viable because then you cannot run the all services together since things like auth become necessary AFAIK; and running them together was and will be necessary to check logs in different scenarios. I think there's also advantage to the default being the behaviour we have in production (where practical) because it makes replicating production behaviour simpler. I originally made the change because when I did the JSON work I seem to remember it being more onerous than necessary to get the logs as per prod locally. 

The change as Laurence has it seems simple to toggle local dev between json or std 👍  (I'm not actually working this morning because I'm still unwell)

Reply:
Here's an alternative solution; we move all the logging related code to a new initializer in `config/initializers/logging.rb`, and wrap everything in an `unless Rails.env.dev?` block...

```
# config/initializers/logging.rb

unless Rails.env.development?
  ### LOGGING CONFIGURATION ###
...
end
```

then we don't need to change `environments/development.rb` to get the Rails defaults, but we get the centralised configuration.

How does that sound as a compromise?

Reply:
That seems like a pretty neat solution to me (but I don't mind the current implementation either)

Reply:
Yeah that sounds like a good alternative.  We can also add some unit tests for that initializer to make sure its doing what we expect it to do.

Reply:
I preferred your original implementation using the `config/environments/*.rb` because to me it keeps environmental differences in limited known places and seems like that's what they're intended for. With the suggested alternative and similar types of switches based upon `Rails.env.development` throughout our codebase I think it'll become harder to see whats being controlled where and what the substantial differences are between environments. Although this is a simple and rather innocuous change to our logging it seems this discussion is more about the pattern and one which we might follow in future.

However, if the new suggestion is the least surprising and simple to understand and modify for the majority then I won't stand in the way 😄

Reply:
I've decided we should go with the second option as to me it's clearer because we know it only applies to non-production like env. We should add the test as Al suggests. We already have switches based on env vars in some of the other Rails configs. Looking at each env config to look for how the logging is configured to me is more confusing that seeing a file at a level above named `logging.rb`. Each env file can still override that configuration I believe.

Reply:
Sounds good and thanks for giving time to the conversation and arriving at a decision 👍 . I think personal preference comes down to which end of the stick you're looking at (and no doubt prior Rails experience). I can see this works for finding configuration differences of a particular component, in this case logging, by looking at its initializer. As someone who's been spending more time in Terraform (and less Rails) my focus was more on making it simpler to find the broader differences between development and the default (which ideally aligns with production) by just looking at `config/environments/development.rb`.

Onwards and thanks for making this improvement Laurence! 🚀

Reply:
Okay, I've pushed a commit that moves the logging code to a separate file, and adds tests. Please re-review all!